src/Type/Definition/CustomScalarType.php

```diff
@@ -54,16 +61,18 @@ public function assertValid()
 
         Utils::invariant(
             isset($this->config['serialize']) && is_callable($this->config['serialize']),
-            "{$this->name} must provide \"serialize\" function. If this custom Scalar " .
+            sprintf('%s must provide "serialize" function. If this custom Scalar ', $this->name) .
             'is also used as an input type, ensure "parseValue" and "parseLiteral" ' .
             'functions are also provided.'
         );
-        if (isset($this->config['parseValue']) || isset($this->config['parseLiteral'])) {
-            Utils::invariant(
-                isset($this->config['parseValue']) && isset($this->config['parseLiteral']) &&
-                is_callable($this->config['parseValue']) && is_callable($this->config['parseLiteral']),
-                "{$this->name} must provide both \"parseValue\" and \"parseLiteral\" functions."
-            );
+        if (! isset($this->config['parseValue']) && ! isset($this->config['parseLiteral'])) {
+            return;
```

Comment:
I am curious if this style of logic is recommended by the coding standard (when you return or continue and leave the main logic at the bottom of the function / loop)?

Reply:
Yes, it's called early exit. It emphatises the actual logic so it is not unecessarily nested and all distracting conditions is executed before it.

The code is then easier to read as the possitive path is on the first level in the function. Typical example https://github.com/doctrine/dbal/pull/2937/commits/6013259628fae7ac1e5cddebce8f83528b9031a6

Reply:
Don't see any advantage of not using early return, only that one that the code somehow mirrors that reference implementation, that would make sense.

For debugging, well, ¯\_(ツ)_/¯  
Since I use xdebug, I never dump values. Of course it's about personal preference but code should not be influenced by that preference.

Also, adding some more reference  
https://www.slideshare.net/guilhermeblanco/php-for-adults-clean-code-and-object-calisthenics 39+  
https://www.slideshare.net/guilhermeblanco/object-calisthenics-applied-to-php 9  
https://github.com/jupeter/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-1  
https://github.com/jupeter/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-2

It's your call, we can exclude that sniff ofc.

Where can I find the reference code BTW? Thanks!

Reply:
So it is a sniff from the coding standard, right?

[Reference GraphQL implementation](https://github.com/graphql/graphql-js)

Reply:
Yes

Reply:
Kk, so I guess that's all.

Thanks for the link btw